hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -58,44 +71,84 @@ public HoodieCommitArchiveLog(HoodieWriteConfig config, FileSystem fs) {
             .getArchiveLogPath(config.getBasePath() + "/" + HoodieTableMetaClient.METAFOLDER_NAME);
     }
 
+    private HoodieLogFormat.Writer openWriter() {
+        try {
+            if(this.writer == null) {
+                return HoodieLogFormat.newWriterBuilder()
+                        .onParentPath(archiveFilePath.getParent())
+                        .withFileId(archiveFilePath.getName())
+                        .withFileExtension(HoodieLogFile.DELTA_EXTENSION)
+                        .withFs(fs)
+                        .overBaseCommit("").build();
+            } else {
+                return this.writer;
+            }
+        } catch(InterruptedException | IOException e) {
+            throw new HoodieException("Unable to initialize HoodieLogFormat writer", e);
+        }
+    }
+
+    private void close() {
+        try {
+            if(this.writer != null) {
+                this.writer.close();
+            }
+        } catch(IOException e) {
+            throw new HoodieException("Unable to close HoodieLogFormat writer", e);
+        }
+    }
+
     /**
      * Check if commits need to be archived. If yes, archive commits.
      */
     public boolean archiveIfRequired() {
-        List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
-        boolean success = true;
-        if (commitsToArchive.iterator().hasNext()) {
-            log.info("Archiving commits " + commitsToArchive);
-            archive(commitsToArchive);
-            success = deleteInstants(commitsToArchive);
-        } else {
-            log.info("No Commits to archive");
+        try {
+            List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
+            boolean success = true;
+            if (commitsToArchive.iterator().hasNext()) {
+                this.writer = openWriter();
+                log.info("Archiving commits " + commitsToArchive);
+                archive(commitsToArchive);
+                success = deleteInstants(commitsToArchive);
+            } else {
+                log.info("No Commits to archive");
+            }
+            return success & deleteOtherInstants();
+        } catch(Exception e) {
+            throw new HoodieException("Unable to archive commits", e);
```

Comment:
Why catch if you are just wrapping it over. Just do try finally - if you want to reliably close?